client/src/app/pages/applications/generate-assets-wizard/step-select-target-profile.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import { useForm, useWatch, UseFormReturn } from "react-hook-form";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+import {
+  Form,
+  TextContent,
+  Text,
+  FormGroup,
+  Radio,
+} from "@patternfly/react-core";
+
+import { Archetype, TargetProfile } from "@app/api/models";
+import { DecoratedApplication } from "../useDecoratedApplications";
+import { useFetchArchetypes } from "@app/queries/archetypes";
+
+interface TargetProfileFormValues {
+  selectedProfile?: TargetProfile;
+}
+
+export interface TargetProfileState {
+  selectedProfile?: TargetProfile;
+  isValid: boolean;
+}
+
+const useTargetProfileStateChangeHandler = (
+  form: UseFormReturn<TargetProfileFormValues>,
+  onTargetProfileChanged: (profileState: TargetProfileState) => void
+) => {
+  const {
+    control,
+    formState: { isValid },
+  } = form;
+
+  const watchedValues = useWatch({
+    control,
+    name: ["selectedProfile"],
+  });
+
+  const profileState = React.useMemo((): TargetProfileState => {
+    const [selectedProfile] = watchedValues;
+    return {
+      selectedProfile,
+      isValid,
+    };
+  }, [watchedValues, isValid]);
+
+  React.useEffect(() => {
+    onTargetProfileChanged(profileState);
+  }, [onTargetProfileChanged, profileState]);
+};
+
+export const SelectTargetProfile: React.FC<{
+  applications: DecoratedApplication[];
+  onTargetProfileChanged: (profile: TargetProfile) => void;
+  initialTargetProfile?: TargetProfile;
+}> = ({ applications, onTargetProfileChanged, initialTargetProfile }) => {
+  const { t } = useTranslation();
+  const { archetypes } = useFetchArchetypes();
```

Comment:
1. consider pre-fetching data - this is the first step but the others might benefit from "hot" cache
2. some loading state handling would be nice

Reply:
Turns out I already had this info in the DecoratedApplication!  No extra fetches needed.